src/compile/facet.ts

```diff
@@ -108,14 +108,14 @@ export class FacetModel extends Model {
     }, {});
   }
 
-  private _initAxis(facet: Facet, config: Config, child: Model): Dict<Axis> {
+  private _initAxis(facet: Facet, child: Model): Dict<Axis> {
     const model = this;
     return [ROW, COLUMN].reduce(function(_axis, channel) {
       if (facet[channel]) {
         const axisSpec = facet[channel].axis;
         if (axisSpec !== false) {
           const modelAxis = _axis[channel] = extend({},
-            config.facet.axis,
+            defaultFacetAxisConfig,
```

Comment:
In any case, now you have those rules, why don't you fix your code here (and add test).